src/view/treeNodes/pullRequestNode.ts

```diff
@@ -504,265 +557,188 @@ export class PRNode extends TreeNode {
-			added: [],
-			changed: changedCommentThreads,
-			removed: []
-		});
-	}
-
-	async deleteReaction(document: vscode.TextDocument, comment: vscode.Comment, reaction: vscode.CommentReaction) {
-		const fileChange = this.findMatchingFileNode(document.uri);
-		let matchedRawComment = fileChange.comments.find(cmt => String(cmt.id) === comment.commentId);
-
-		if (!matchedRawComment) {
-			throw new Error('Unable to find matching comment');
-		}
-
-		await this._prManager.deleteCommentReaction(this.pullRequestModel, matchedRawComment.graphNodeId, reaction);
-		const params = fromPRUri(document.uri);
-		let comments = await this._prManager.getPullRequestComments(this.pullRequestModel);
-		let changedCommentThreads = commentsToCommentThreads(fileChange, comments.filter(cmt => cmt.path === fileChange.fileName && cmt.position !== null), params!.isBase);
-
-		this._onDidChangeCommentThreads.fire({
-			added: [],
-			changed: changedCommentThreads,
-			removed: []
-		});
-	}
+	// #endregion
+	// async addReaction(document: vscode.TextDocument, comment: vscode.Comment, reaction: vscode.CommentReaction) {
+	// 	const fileChange = this.findMatchingFileNode(document.uri);
+	// 	if (!fileChange) {
+	// 		throw new Error('Unable to find matching file');
+	// 	}
+
+	// 	let matchedRawComment = fileChange.comments.find(cmt => String(cmt.id) === comment.commentId);
+
+	// 	if (!matchedRawComment) {
+	// 		throw new Error('Unable to find matching comment');
+	// 	}
+
+	// 	await this._prManager.addCommentReaction(this.pullRequestModel, matchedRawComment.graphNodeId, reaction);
+	// 	const params = fromPRUri(document.uri);
+	// 	let comments = await this._prManager.getPullRequestComments(this.pullRequestModel);
+	// 	let changedCommentThreads = commentsToCommentThreads(fileChange, comments.filter(cmt => cmt.path === fileChange.fileName && cmt.position !== null), params!.isBase);
+
+		// this._onDidChangeCommentThreads.fire({
+		// 	added: [],
+		// 	changed: changedCommentThreads,
+		// 	removed: []
+		// });
+	// }
+
+	// async deleteReaction(document: vscode.TextDocument, comment: vscode.Comment, reaction: vscode.CommentReaction) {
+	// 	const fileChange = this.findMatchingFileNode(document.uri);
+	// 	let matchedRawComment = fileChange.comments.find(cmt => String(cmt.id) === comment.commentId);
+
+	// 	if (!matchedRawComment) {
+	// 		throw new Error('Unable to find matching comment');
+	// 	}
+
+	// 	await this._prManager.deleteCommentReaction(this.pullRequestModel, matchedRawComment.graphNodeId, reaction);
+	// 	const params = fromPRUri(document.uri);
+	// 	let comments = await this._prManager.getPullRequestComments(this.pullRequestModel);
+	// 	let changedCommentThreads = commentsToCommentThreads(fileChange, comments.filter(cmt => cmt.path === fileChange.fileName && cmt.position !== null), params!.isBase);
+
+		// this._onDidChangeCommentThreads.fire({
+		// 	added: [],
+		// 	changed: changedCommentThreads,
+		// 	removed: []
+		// });
```

Comment:
When the comment controller is disposed, are all of the comment threads associated with it also removed?

Reply:
yes, they will all be disposed first before the controller is disposed.

Reply:
yes, the way we refresh tree nodes should be improved. Previously it's pull mode so it's okay to dipose all tree nodes and recreate them.